nucleus/admin/monitor/src/main/java/fish/payara/admin/monitor/cli/SetMonitoringConfiguration.java

```diff
@@ -0,0 +1,259 @@
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.admin.monitor.cli;
+
+import com.sun.enterprise.config.serverbeans.Config;
+import com.sun.enterprise.config.serverbeans.Domain;
+import com.sun.enterprise.config.serverbeans.MonitoringService;
+import com.sun.enterprise.util.SystemPropertyConstants;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.I18n;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.CommandLock;
+import org.glassfish.api.admin.CommandRunner;
+import org.glassfish.api.admin.CommandRunner.CommandInvocation;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.ParameterMap;
+import org.glassfish.api.admin.RestEndpoint;
+import org.glassfish.api.admin.RestEndpoints;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+import org.jvnet.hk2.config.ConfigSupport;
+import org.jvnet.hk2.config.TransactionFailure;
+
+import static org.glassfish.api.ActionReport.ExitCode.FAILURE;
+import static org.glassfish.api.ActionReport.ExitCode.SUCCESS;
+import static org.glassfish.api.ActionReport.ExitCode.WARNING;
+
+/**
+ * @author Susan Rai
+ */
+@Service(name = "set-monitoring-configuration")
+@PerLookup
+@CommandLock(CommandLock.LockType.NONE)
+@I18n("set.monitoring.configuration")
+@ExecuteOn(RuntimeType.DAS)
+@TargetType(value = {CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.CLUSTER, CommandTarget.CLUSTERED_INSTANCE, CommandTarget.CONFIG, CommandTarget.DEPLOYMENT_GROUP})
+@RestEndpoints({
+    @RestEndpoint(configBean = Domain.class,
+            opType = RestEndpoint.OpType.POST,
+            path = "set-monitoring-configuration",
+            description = "This command can be used to configure all the monitoring related configuration")
+})
+public class SetMonitoringConfiguration implements AdminCommand {
+
+    private static final Logger logger = Logger.getLogger(SetMonitoringConfiguration.class.getName());
+
+    @Inject
+    private Target targetUtil;
+
+    @Param(optional = true, alias = "monitoringEnabled")
+    private Boolean monitoringServiceEnabled;
```

Comment:
Really strong preference for an `--enabled` option here. Is there a reason I'm missing that this isn't appropriate?

Reply:
It was done to avoid the ambiguity as the command can be used to enable 3 different monitoring related services.

Reply:
I would say it's fine to have `--enabled` as the alias. 
So if being specific it would be `--monitoringserviceenabled` (which is what gets shown when doing `set-monitoring-configuration --help`) but it will still allow `--enabled`

Reply:
And yes I realise I've changed my mind :P

Reply:
I think the problem for me is that I would expect `set-monitoring-configuration --enabled=false` and `set-monitoring-configuration --monitoringEnabled` to do exactly the same thing. I'm undecided whether that's at all relevant to this issue though, it's perhaps more of an issue with the monitoring service labels in general. With that in mind, Andy's solution seems to settle things for now